kratos/input_output/vtk_output.cpp

```diff
@@ -120,104 +127,71 @@ void VtkOutput::WriteConditionsAndElements(const ModelPart &rModelPart, std::ofs
     // write cells header
     rFileStream << "\nCELLS " << local_mesh.NumberOfConditions() + local_mesh.NumberOfElements() << " " << mVtkCellListSize << "\n";
 
-    // write elements
-    for (const auto& elem : local_mesh.Elements())
-    {
-        const auto& elem_geometry = elem.GetGeometry();
-        const unsigned int number_of_nodes = elem_geometry.size();
-
-        WriteScalarDataToFile((unsigned int)number_of_nodes, rFileStream);
-        for (const auto& node : elem_geometry){
-            if (mFileFormat == VtkOutput::FileFormat::VTK_ASCII) rFileStream << " ";
-            int id = mKratosIdToVtkId[node.Id()];
-            WriteScalarDataToFile((int)id, rFileStream);
-        }
-        if (mFileFormat == VtkOutput::FileFormat::VTK_ASCII) rFileStream << "\n";
-    }
+    WriteConnectivity<Element> (local_mesh.Elements().GetContainer(), rFileStream);
+    WriteConnectivity<Condition> (local_mesh.Conditions().GetContainer(), rFileStream);
+}
 
+template <typename TEntity>
+void VtkOutput::WriteConnectivity(const typename PointerVectorSet<TEntity, IndexedObject>::ContainerType& rContainer, std::ofstream& rFileStream)
+{
     // write Conditions
-    for (const auto& condition : local_mesh.Conditions())
+    for (const auto& entity : rContainer)
     {
-        const auto& condition_geometry = condition.GetGeometry();
-        const unsigned int number_of_nodes = condition_geometry.size();
+        const auto& entity_geometry = entity->GetGeometry();
+        const unsigned int number_of_nodes = entity_geometry.size();
 
         WriteScalarDataToFile((unsigned int)number_of_nodes, rFileStream);
-        for (const auto& node : condition_geometry){
+        for (const auto& node : entity_geometry){
             if (mFileFormat == VtkOutput::FileFormat::VTK_ASCII) rFileStream << " ";
             int id = mKratosIdToVtkId[node.Id()];
             WriteScalarDataToFile((int)id, rFileStream);
         }
         if (mFileFormat == VtkOutput::FileFormat::VTK_ASCII) rFileStream << "\n";
     }
-
 }
 
 void VtkOutput::WriteConditionAndElementTypes(const ModelPart &rModelPart, std::ofstream& rFileStream)
 {
+
     const auto& local_mesh = rModelPart.GetCommunicator().LocalMesh();
     // write cell types header
     rFileStream << "\nCELL_TYPES " << local_mesh.NumberOfConditions() + local_mesh.NumberOfElements() << "\n";
+    WriteCellType<Element> (local_mesh.Elements().GetContainer(), rFileStream);
+    WriteCellType<Condition> (local_mesh.Conditions().GetContainer(), rFileStream);
```

Comment:
And this would be 
~~~c
WriteCellType(local_mesh.Elements(), rFileStream);
~~~